File: src/SkyBlock/enchants/armor/Enlighten.php
<?php


namespace SkyBlock\enchants\armor;


use pocketmine\player\Player;

class Enlighten extends BaseArmorEnchant {

    static int $id = 124;

    /**
     * @param Player $attacker
     * @param Player $victim
     * @param int    $enchantmentLevel
     */
    public function onActivation(Player $attacker, Player $victim, int $enchantmentLevel) : void {
        $this->sendActivation($victim, "§bEnlighten §aActivated!");
        $victim->setHealth($victim->getHealth() + ($enchantmentLevel / 40));
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/armor/Enlighten.java
package dev.skyblock.enchants.armor;
import dev.skyblock.enchants.BaseArmorEnchant;
import org.bukkit.attribute.Attribute;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public class Enlighten extends BaseArmorEnchant {
    public static final int ID = 124;
    public Enlighten() { super(ID, "Enlighten", 15); }
    @Override
    public boolean isApplicableTo(Player player, int level) { return java.util.concurrent.ThreadLocalRandom.current().nextInt(1, 26) == 1; }
    @Override
    public void onActivation(Player victim, EntityDamageByEntityEvent ev, int level) {
        victim.sendMessage("§bEnlighten §aActivated!");
        double heal = level / 40.0;
        double maxHealth = victim.getAttribute(Attribute.MAX_HEALTH).getValue();
        victim.setHealth(Math.min(maxHealth, victim.getHealth() + heal));
    }
}
